README.md

```diff
@@ -149,23 +150,68 @@ the creation of a new management pack project.
+   PNG format and 256x256 pixels. An icon file can be added later by copying the icon to the root project directory and
    setting the value of the `"pak_icon"` key to the icon's file name in the `manifest.txt` file.
 
-8. `Select a language for the adapter. Supported languages are [...]`
-
-   Supported languages are listed. Once selected, the project will be generated, including a template adapter in the
-   selected language. The template adapter collects several objects and metrics from the container that the adapter 
-   is running in, and can be used as a starting point for creating a new adapter.
-
-For complete documentation of the `mp-init` tool see the [MP Initialization Tool Documentation](doc/mp-init.md).
+[//]: # (TODO: Add this section back when we support it)
+[//]: # (8. `Select a language for the adapter`)
+[//]: # (  selected language. The template adapter collects several objects and metrics from the container that the adapter)
+[//]: # (  Once selected, the project will be generated, including a template adapter in the)
+[//]: # (  is running in, and can be used as a starting point for creating a new adapter.)
+
+For complete documentation of the `mp-init` tool including an overview of its output, see the [MP Initialization Tool Documentation](doc/mp-init.md).
+
+### Template Project
+Every new project creates a file system that has the basic project structure required to develop and build a Management Pack.
+Each file and directory are discussed in depth in the [mp-init](doc/mp-init.md) documentation. `app/adapter.py` is the adapter's
+entry point and the best starting point. `adapter.py` is a template adapter that collects several objects and metrics from the
+container in which the adapter is running; use the template as a starting point for creating a new adapter. The template adapter 
+has comments throughout its code that explain what the code does and how to turn it into your adapter. The methods inside the adapter
+template are required. Modify the code inside the methods to generate the desired adapter. Each method represents a single request,
+and it can be tested individually using `mp-test`, which is covered in the following section. The adapter is stateless; therefore,
+the methods don't store any data for use in later method calls. Each method is used for a different function as described below:
+
+- test(adapter_instance):
+  Performs a test connection using the information given to the adapter_instance to verify the adapter has been set up properly.
+  A typical test connection will generally consist of:
+
+     1. Read identifier values from adapter_instance that are required to connect to the target(s)
+     2. Connect to the target(s), and retrieve some sample data
+     3. Disconnect cleanly from the target (ensure this happens even if an error occurs)
+     4. If any of the above failed, return an error, otherwise pass.
```

Comment:
```suggestion
     3. If any of the above failed, return an error, otherwise pass.
     4. Disconnect cleanly from the target (ensure this happens even if an error occurs)
```